internal/utils.go

```diff
@@ -252,13 +253,20 @@ func getPolicyTemplates(policyConf *types.PolicyConfig) ([]map[string]interface{
 				// put all objTemplate with manifest into single consolidated objectTemplates
 				objectTemplates = append(objectTemplates, objTemplate)
 			} else {
+				// Do not append a number to configName when it is used for the first time.
+				configName := policyConf.Manifests[i].Name
+				if j != 0 && configName != "" {
+					configName = fmt.Sprintf("%s%d", policyConf.Manifests[i].Name, j+1)
+				}
+
 				// casting each objTemplate with manifest to objectTemplates type
 				// build policyTemplate for each objectTemplates
 				policyTemplate := buildPolicyTemplate(
 					policyConf,
 					len(policyTemplates)+1,
 					[]map[string]interface{}{objTemplate},
 					&policyConf.Manifests[i].ConfigurationPolicyOptions,
+					configName,
```

Comment:
(Optional) I might prefer we pass the parent object to the function and set `policyConf.Manifests[i].Name` in `ApplyDefaults()`: 
```suggestion
					&policyConf.Manifests[i],
```
Then in the conditional above, we detect whether `policyConf.name == Manifests[i].Name` to determine whether it's been overridden, and then set `Manifests[i].Name` directly with the appropriate index if necessary and use that value directly for the name in `buildPolicyTemplate()`. This would also properly handle the case where the same name is given for the manifest and the policy.

Reply:
I need a` j` (When it has multiple manifest in the Path file) to attach numbers to configurationpolicy name

Reply:
Right, I'm proposing the conditional above these lines would use `j` and set the generator struct directly before passing it to `buildPolicyTemplate()`, something like this to detect whether the name was overridden:
```
if j != 0 && policyConf.name != policyConf.Manifests[i].Name {
	policyConf.Manifests[i].Name = fmt.Sprintf("%s%d", policyConf.Manifests[i].Name, j+1)
}
```

Reply:
I refactored. I added this func
```
func getConfigurationPolicyName(policyConf *types.PolicyConfig, manifestGroupIndex, policyNum,
	manifestIndex int, useManifestName bool,
) string {
	// Do not append a number to configName when it is used for the first time.
	var configName string

	if useManifestName && policyConf.Manifests[manifestGroupIndex].Name != "" {
		// If the user does not added Manifests[manifestGroupIndex].Name
		// then return policyConf.Name as a configurationPolicyName
		configName = policyConf.Manifests[manifestGroupIndex].Name
		if manifestIndex > 0 && configName != "" {
			return fmt.Sprintf("%s%d", configName, manifestIndex+1)
		}
	}

	if configName == "" {
		configName = policyConf.Name
		if policyNum > 1 {
			return fmt.Sprintf("%s%d", configName, policyNum)
		}
	}

	return configName
}
```